layouts/partials/page/list.html

```diff
@@ -1,4 +1,18 @@
-{{.Content}}
+{{ if (ne (len .Content) 0) }}
+    {{ $roles := .Params.roles | default "All Roles" }}
+    {{ $uid := .File.UniqueID }}
+    {{ $articleId := .Params.id | default .File.Path }}
+    {{ $articleLink := (printf "%v#%v" .Parent.Page.RelPermalink ) }}
+    {{ $slug := (partial "common/slug" .) }}
+    <div class="article child" data-roles="{{ $roles }}" id="{{ $uid }}" data-link="{{ $articleLink }}" permalink="{{ .Page.RelPermalink }}">
```

Comment:
Using a difference class will require work on frontend.

We can extend the class name to make it more declarative: eg `class="article child index"`